File: golang/sieve_of_eratosthenes.go
package main

import (
	"fmt"
	"math"
)

// Sieve of eratosthenes
// Algorithm to generate prime numbers till n
func sieve(n int) {
	var arr = make([]int, n+1)
	for i := 0; i <= n; i++ {
		arr[i] = i
	}
	for i := 2; i <= int(math.Sqrt(float64(n))); i++ {
		if arr[i] != 0 {
			j := i * i
			for j <= n {
				arr[j] = 0
				j = j + i
			}
		}
	}
	var res = make([]int, n+1)
	j := 0
	for i := 2; i <= n; i++ {
		if arr[i] != 0 {
			res[j] = arr[i]
			j++
		}
	}
	for _, v := range res {
		if v != 0 {
			fmt.Printf("%d ", v)
		}
	}
	fmt.Println("")
}

func main() {
	sieve(25)
}

File: python/sieve_of_eratosthenes.py
''' Sieve of Eratosthenes '''

import math
import sys


def sieve(num):
    ''' Algorithm to generate list of prime numbers till num '''
    arr = range(0, num + 1)
    for idx in range(2, int(math.sqrt(num)) + 1):
        if arr[idx] != 0:  # idx hasnt been eliminated on previous passes
            j = idx * idx
            while j <= num:
                arr[j] = 0
                j = j + idx
    result = []
    for idx in range(2, num + 1):
        if arr[idx] != 0:
            result.append(arr[idx])
    return result


if __name__ == '__main__':
    N = 25  # default value
    if len(sys.argv) == 2:
        N = int(sys.argv[1])
    print sieve(N)
